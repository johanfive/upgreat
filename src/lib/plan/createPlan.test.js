const createPlan = require('./createPlan')

const bluebird = require('bluebird')
jest.mock('bluebird')

it('should return {plan, missingPeers, errors}', async () => {
  bluebird.map.mockImplementationOnce(() =>
    Promise.resolve([
      {
        name: '404-pkg',
        err: 404,
        peers: [],
      },
      {
        name: 'p1',
        diff: 'minor',
        peers: ['p2'],
      },
      {
        name: 'p2',
        diff: 'major',
        peers: ['nonMissingPeer'],
      },
      {
        name: 'p3',
        diff: 'patch',
        peers: ['p2', 'missingPeer'],
      },
      {
        name: 'nonMissingPeer',
        diff: null,
        peers: [],
      },
    ]),
  )

  await expect(
    createPlan(
      [
        { name: '404-pkg', version: '4.0.4' },
        { name: 'p1', version: '1.1.1' },
        { name: 'p2', version: '2.2.2' },
        { name: 'p3', version: '3.3.3' },
      ],
      {
        nonMissingPeer: '4.4.4',
      },
    ),
  ).resolves.toMatchSnapshot()
})

it('should work with no args', async () => {
  bluebird.map.mockImplementationOnce(() => Promise.resolve([]))

  await expect(createPlan()).resolves.toMatchSnapshot()
})
