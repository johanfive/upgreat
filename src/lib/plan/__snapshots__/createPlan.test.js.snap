// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return {plan, missingPeers, errors} 1`] = `
Object {
  "errors": Array [
    Object {
      "err": 404,
      "name": "404-pkg",
      "peers": Array [],
    },
  ],
  "missingPeers": Array [
    "missingPeer",
  ],
  "plan": Array [
    Object {
      "diff": null,
      "name": "nonMissingPeer",
      "peers": Array [],
    },
    Object {
      "diff": "major",
      "name": "p2",
      "peers": Array [
        "nonMissingPeer",
      ],
    },
    Object {
      "diff": "patch",
      "name": "p3",
      "peers": Array [
        "p2",
      ],
    },
    Object {
      "diff": "minor",
      "name": "p1",
      "peers": Array [
        "p2",
      ],
    },
  ],
}
`;

exports[`should work with no args 1`] = `
Object {
  "errors": Array [],
  "missingPeers": Array [],
  "plan": Array [],
}
`;
